### PR TITLE
Handle lookup returning empty array of users.

### PR DESCRIPTION
**Pull request by yuchenshi**
### Discussion

This is a follow up to #1078 and improves the detection to be more robust for detecting no users returned.

Previously, it depends on `.users` not being present (instead of an empty array) when there are no users. This is not explicitly part of the [API documentation](https://cloud.google.com/identity-platform/docs/reference/rest/v1/projects.accounts/lookup) or [contract](https://cloud.google.com/identity-platform/docs/reference/rest/v1/GetAccountInfoResponse) and we probably shouldn't depend on it.

### Testing

* New tests added and existing tests updated for the responseValidator.

### API Changes

No public API changes included.

### CODE REVIEW

File: test/unit/auth/auth-api-request.spec.ts [hunk 1]
```diff
@@ -369,17 +369,23 @@ describe('FIREBASE_AUTH_GET_ACCOUNT_INFO', () => {
...
     const responseValidator = FIREBASE_AUTH_GET_ACCOUNT_INFO.getResponseValidator();
     it('should succeed with users returned', () => {
-      const validResponse: object = { users: [] };
+      const validResponse: object = { users: [{ localId: 'foo' }] };
       expect(() => {
         return responseValidator(validResponse);
       }).not.to.throw();
     });
-    it('should fail when users is not returned', () => {
+    it('should fail when users is empty', () => {
```
Comment (bojeil-google) [inline test/unit/auth/auth-api-request.spec.ts:377]: nit: this test description and the one below sounds similar. Let's rephrase to be clearer:
`should fail when the response object is empty`

File: test/unit/auth/auth-api-request.spec.ts [hunk 2]
```diff
@@ -369,17 +369,23 @@ describe('FIREBASE_AUTH_GET_ACCOUNT_INFO', () => {
...
       const invalidResponse = {};
       expect(() => {
         responseValidator(invalidResponse);
       }).to.throw();
     });
+    it('should fail when users is not returned', () => {
```
Comment (bojeil-google) [inline test/unit/auth/auth-api-request.spec.ts:383]: `should fail when the response object has an empty list of users`

**Review by bojeil-google — CHANGES_REQUESTED** [review]
Just 2 nits

File: test/unit/auth/auth-api-request.spec.ts (hunk 1 above)
Comment (yuchenshi) [reply test/unit/auth/auth-api-request.spec.ts]: Done! Thanks for the suggestion!

File: test/unit/auth/auth-api-request.spec.ts (hunk 2 above)
Comment (yuchenshi) [reply test/unit/auth/auth-api-request.spec.ts]: Done! Thanks for the suggestion!

**Review by bojeil-google — APPROVED** [review]

**Review by hiranya911 — APPROVED** [review]
LGTM

**Pull request merged**